gusto/physics.py

```diff
@@ -78,31 +78,30 @@ def __init__(self, state):
         w_sat = (w_sat1 /
                  (p * exp(w_sat2 * (T - T_0) / (T - w_sat3)) - w_sat4))
 
-        # correct w_sat to be positive
-        w_sat = interpolate(conditional(w_sat > 0, w_sat, 999.0), V)
-
         # make appropriate condensation rate
-        cond_rate = ((self.water_v - w_sat) /
-                     (dt * (1.0 + ((L_v ** 2.0 * w_sat) /
-                                   cp * R_v * T ** 2.0))))
+        dot_r_cond = ((self.water_v - w_sat) /
+                      (dt * (1.0 + ((L_v ** 2.0 * w_sat) /
+                                    (cp * R_v * T ** 2.0)))))
 
         # adjust cond rate so negative concentrations don't occur
-        cond_rate = interpolate(conditional(cond_rate < 0,
-                                conditional(-cond_rate > self.water_c / dt,
-                                            -self.water_c / dt,
-                                            cond_rate),
-                                conditional(cond_rate > self.water_v / dt,
-                                            self.water_v / dt,
-                                            cond_rate)), V)
-
-        self.water_v_new = Interpolator(self.water_v - dt * cond_rate, V)
-        self.water_c_new = Interpolator(self.water_c + dt * cond_rate, V)
+        self.lim_cond_rate = Interpolator(conditional(dot_r_cond < 0,
+                                                      conditional(-dot_r_cond > self.water_c / dt,
+                                                                  -self.water_c / dt,
+                                                                  dot_r_cond),
+                                                      conditional(dot_r_cond > self.water_v / dt,
+                                                                  self.water_v / dt,
+                                                                  dot_r_cond)), Vt)
```

Comment:
Yep, I think you're exactly right about both of these. I don't understand why I didn't do it that way originally...